pkg/sdk/testint/setup_integration_test.go

```diff
@@ -31,27 +32,59 @@ func setup() {
 	err := itc.initialize()
 	if err != nil {
 		log.Printf("Integration test context initialisation failed with %s\n", err)
+		cleanup()
 		os.Exit(1)
 	}
 }
 
 func cleanup() {
 	log.Println("Running integration tests cleanup")
+	if itc.databaseCleanup != nil {
+		defer itc.databaseCleanup()
```

Comment:
I think defer is not needed in this context

Reply:
It will be when we have more resources to clean here.